lib/content-services/src/lib/tag/tags-creator/tags-creator.component.ts

```diff
@@ -0,0 +1,379 @@
+        this.tagNameControlVisibleChange.emit(this.tagNameControlVisible);
+    }
+
+    addTag(): void {
+        if (!this._typing && !this.tagNameControl.invalid) {
+            this.tags.push(this.tagNameControl.value.trim());
+            this.hideNameInput();
+            this.tagNameControl.setValue('');
+            this.checkScrollbarVisibility();
+            this.tagNameControl.markAsUntouched();
+            this.tagsChange.emit(this.tags);
+        }
+    }
+
+    removeTag(tag: string): void {
+        this.removeTagFromArray(this.tags, tag);
+        this.tagNameControl.updateValueAndValidity();
+        this.updateExistingTagsListOnRemoveFromTagsToConfirm(tag);
+        this.checkScrollbarVisibility();
+        this.tagsChange.emit(this.tags);
+    }
+
+    addExistingTagToTagsToAssign(change: MatSelectionListChange): void {
+        const selectedTag: TagEntry = change.options[0].value;
+        this.tags.push(selectedTag.entry.tag);
+        this.removeTagFromArray(this.existingTags, selectedTag);
+        this.tagNameControl.updateValueAndValidity();
+        this.exactTagSet$.next();
+        this.tagsChange.emit(this.tags);
+    }
+
+    isOnlyCreateMode(): boolean {
+        return this.mode === TagsCreatorMode.CREATE;
+    }
+
+    private onTagNameControlValueChange(name: string): void {
+        this.tagNameControl.markAsTouched();
+        this.loadTags(name);
+    }
+
+    private loadTags(name: string) {
+        if (name) {
+            forkJoin({
+                exactResult: this.tagService.findTagByName(name),
+                searchedResult: this.tagService.searchTags(name, DEFAULT_TAGS_SORTING, false, 0, this.existingTagsListLimit)
+            }).pipe(
+                takeUntil(this.cancelExistingTagsLoading$),
+                finalize(() => (this._typing = false))
+            ).subscribe(({ exactResult, searchedResult }: {
+                exactResult: TagEntry;
+                searchedResult: TagPaging;
+            }) => {
+                if (exactResult) {
+                    this.existingExactTag = exactResult;
+                    this.removeExactTagFromSearchedResult(searchedResult);
+                    searchedResult.list.entries.unshift(exactResult);
+                } else {
+                    this.existingExactTag = null;
+                }
+
+                this._initialExistingTags = searchedResult.list.entries;
+                this.excludeAlreadyAddedTags(this._initialExistingTags);
+                this.exactTagSet$.next();
+                this._existingTagsLoading = false;
+            }, () => {
+                this.notificationService.showError('TAG.TAGS_CREATOR.ERRORS.FETCH_TAGS');
+                this._existingTagsLoading = false;
+            });
+        } else {
+            this.existingExactTag = null;
+        }
+    }
+
+    private removeExactTagFromSearchedResult(searchedResult: TagPaging): void {
```

Comment:
Can be removed if we don't need another search for exact tag

Reply:
see answer above, we need separate search :)